perceval/components/abstract_processor.py

```diff
@@ -90,6 +94,10 @@ def parameters(self):
     def clear_parameters(self):
         self._parameters = {}
 
+    def clear_input_and_circuit(self):
```

Comment:
In which workflow do we have to use these clear_XXX methods ?
Is it just to avoid to create a new processor object ?

Reply:
Yes, `clear_input_and_circuit` allows to keep all the `Processor` config and refresh only circuit and input state. It is required to improve the syntax of upcoming code.

`clear_postprocess()` can already be useful during Processor composition with others, as a postprocess function blocks any further composition.

Reply:
It looks strange to have a class named Processor where you can clear the circuit.
Also I prefer to create a new object and copy several parameters than clearing some members of an existing one.

Reply:
In case we have an optimisation process that tries to place components on its own, it can be useful to clear the processor as the number of modes or the input state can change during the optimisation process, as well as the circuit itself.
If you want, you can of course redefine at each iterations the "constants" of your processor, but you can also only change these parameters. This avoids for example to ask the cloud at each iteration the specs of a RemoteProcessor and deserialise them (which is done automatically).